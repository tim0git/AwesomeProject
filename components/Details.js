import React from 'react';
// import PropTypes from 'prop-types';
import {
  //   SafeAreaView,
  //   StyleSheet,
  // ScrollView,
  View,
  Text,
  //   StatusBar,
} from 'react-native';

function Details(props) {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text>Details Screen</Text>
    </View>
  );
}

// Details.propTypes = {};

export default Details;
