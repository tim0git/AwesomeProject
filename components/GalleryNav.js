import React from 'react';
// import PropTypes from 'prop-types'
import {createStackNavigator} from '@react-navigation/stack';
import Gallery from './gallery/Gallery';
import SubGallery from './gallery/SubGallery';

const Stack = createStackNavigator();

function GalleryNav(props) {
  return (
    <Stack.Navigator initialRouteName={Gallery}>
      <Stack.Screen name="Gallery" component={Gallery} />
      <Stack.Screen name="Mountains" component={SubGallery} />
      <Stack.Screen name="Forrest" component={SubGallery} />
      <Stack.Screen name="Beach" component={SubGallery} />
      <Stack.Screen name="City" component={SubGallery} />
    </Stack.Navigator>
  );
}

// GalleryNav.propTypes = {}

export default GalleryNav;
