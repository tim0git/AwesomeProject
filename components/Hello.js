import React from 'react';
// import PropTypes from 'prop-types';
import {View, Text} from 'react-native';

function Hello() {
  return (
    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
      <Text id="welcome-message">Hello World</Text>
    </View>
  );
}

// Hello.propTypes = {};

export default Hello;
