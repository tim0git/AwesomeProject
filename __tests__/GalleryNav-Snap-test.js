import 'react-native';
import React from 'react';
import 'jest-enzyme';
import 'react-native-mock-render/mock';
import MockedStackNavigator from '../__mocks__/MockedStackNavigatior';
import Hello from '../components/Hello';
import Details from '../components/Details';
import GalleryNav from '../components/GalleryNav';

// import {shallow} from 'enzyme';
// import toJson from 'enzyme-to-json';

describe('<GalleryNav > Tab Nav Mock Tests', () => {
  it('MockTabNavigator <Hello > should render correctly', () => {
    const toJSON = render(<MockedStackNavigator component={Hello} />);
    expect(toJSON).toMatchSnapshot();
  });

  it('MockTabNavigator <Details > should render correctly', () => {
    const toJSON = render(<MockedStackNavigator component={Details} />);
    expect(toJSON).toMatchSnapshot();
  });

  it('MockTabNavigator <GalleryNav > should render correctly', () => {
    const toJSON = render(<MockedStackNavigator component={GalleryNav} />);
    expect(toJSON).toMatchSnapshot();
  });
});
