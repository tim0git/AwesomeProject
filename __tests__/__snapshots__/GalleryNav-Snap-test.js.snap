// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GalleryNav > Tab Nav Mock Tests MockTabNavigator <Details > should render correctly 1`] = `
<view
  style="flex:1"
>
  <rncsafeareaprovider
    style="0:[object Object]"
  >
    <view
      pointer-events="box-none"
      style="0:[object Object];1:"
    >
      <view
        importantforaccessibility="auto"
        pointer-events="box-none"
      >
        <view
          pointer-events="box-none"
          style="position:absolute;left:0;right:0;top:0;bottom:0;z-index:0;opacity:1"
        >
          <view
            style="flex:1;shadow-opacity:0.85px;shadow-radius:0;shadow-offset:[object Object];background-color:rgb(255, 255, 255);border-bottom-color:rgb(216, 216, 216);shadow-color:rgb(216, 216, 216)"
          />
        </view>
        <view
          pointer-events="box-none"
          style="height:64px"
        >
          <view
            pointer-events="none"
            style="height:20px"
          />
          <view
            pointer-events="box-none"
            style="flex:1;flex-direction:row;align-items:center;justify-content:center"
          >
            <view
              pointer-events="box-none"
              style="margin-horizontal:16px;opacity:1"
            >
              <text
                accessibilityrole="header"
                numberoflines="1"
                style="font-size:17px;font-weight:600;color:rgb(28, 28, 30)"
              >
                MockedScreen
              </text>
            </view>
          </view>
        </view>
      </view>
    </view>
    <view
      style="flex:1"
    >
      <view
        pointer-events="box-none"
        style="position:absolute;left:0;right:0;top:0;bottom:0"
      >
        <view
          style="opacity:1"
        />
        <view
          gesturevelocityimpact="0.3"
          importantforaccessibility="auto"
          pointer-events="box-none"
          style="position:absolute;left:0;right:0;top:0;bottom:0"
          transitionspec="[object Object]"
        >
          <view
            pointer-events="box-none"
            style="flex:1"
          >
            <view
              style="flex:1;transform:[object Object],[object Object]"
            >
              <view
                pointer-events="none"
                style="position:absolute;shadow-radius:5px;shadow-color:#000;shadow-opacity:0.3px;top:0;bottom:0;width:3px;shadow-offset:[object Object];left:0;background-color:rgb(242, 242, 242)"
              />
              <view
                style="0:[object Object];1:[object Object],"
              >
                <view
                  style="flex:1;flex-direction:column-reverse"
                >
                  <view
                    style="flex:1"
                  >
                    <view
                      style="flex:1;align-items:center;justify-content:center"
                    >
                      <text>
                        Details Screen
                      </text>
                    </view>
                  </view>
                </view>
              </view>
            </view>
          </view>
        </view>
      </view>
    </view>
  </rncsafeareaprovider>
</view>
`;

exports[`<GalleryNav > Tab Nav Mock Tests MockTabNavigator <GalleryNav > should render correctly 1`] = `
<view
  style="flex:1"
>
  <rncsafeareaprovider
    style="0:[object Object]"
  >
    <view
      pointer-events="box-none"
      style="0:[object Object];1:"
    >
      <view
        importantforaccessibility="auto"
        pointer-events="box-none"
      >
        <view
          pointer-events="box-none"
          style="position:absolute;left:0;right:0;top:0;bottom:0;z-index:0;opacity:1"
        >
          <view
            style="flex:1;shadow-opacity:0.85px;shadow-radius:0;shadow-offset:[object Object];background-color:rgb(255, 255, 255);border-bottom-color:rgb(216, 216, 216);shadow-color:rgb(216, 216, 216)"
          />
        </view>
        <view
          pointer-events="box-none"
          style="height:64px"
        >
          <view
            pointer-events="none"
            style="height:20px"
          />
          <view
            pointer-events="box-none"
            style="flex:1;flex-direction:row;align-items:center;justify-content:center"
          >
            <view
              pointer-events="box-none"
              style="margin-horizontal:16px;opacity:1"
            >
              <text
                accessibilityrole="header"
                numberoflines="1"
                style="font-size:17px;font-weight:600;color:rgb(28, 28, 30)"
              >
                MockedScreen
              </text>
            </view>
          </view>
        </view>
      </view>
    </view>
    <view
      style="flex:1"
    >
      <view
        pointer-events="box-none"
        style="position:absolute;left:0;right:0;top:0;bottom:0"
      >
        <view
          style="opacity:1"
        />
        <view
          gesturevelocityimpact="0.3"
          importantforaccessibility="auto"
          pointer-events="box-none"
          style="position:absolute;left:0;right:0;top:0;bottom:0"
          transitionspec="[object Object]"
        >
          <view
            pointer-events="box-none"
            style="flex:1"
          >
            <view
              style="flex:1;transform:[object Object],[object Object]"
            >
              <view
                pointer-events="none"
                style="position:absolute;shadow-radius:5px;shadow-color:#000;shadow-opacity:0.3px;top:0;bottom:0;width:3px;shadow-offset:[object Object];left:0;background-color:rgb(242, 242, 242)"
              />
              <view
                style="0:[object Object];1:[object Object],"
              >
                <view
                  style="flex:1;flex-direction:column-reverse"
                >
                  <view
                    style="flex:1"
                  >
                    <view
                      style="flex:1"
                    >
                      <view
                        style="flex:1"
                      >
                        <view
                          pointer-events="box-none"
                          style="position:absolute;left:0;right:0;top:0;bottom:0"
                        >
                          <view
                            style="opacity:1"
                          />
                          <view
                            gesturevelocityimpact="0.3"
                            importantforaccessibility="auto"
                            pointer-events="box-none"
                            style="position:absolute;left:0;right:0;top:0;bottom:0"
                            transitionspec="[object Object]"
                          >
                            <view
                              pointer-events="box-none"
                              style="flex:1;margin-top:0"
                            >
                              <view
                                style="flex:1;transform:[object Object],[object Object]"
                              >
                                <view
                                  pointer-events="none"
                                  style="position:absolute;shadow-radius:5px;shadow-color:#000;shadow-opacity:0.3px;top:0;bottom:0;width:3px;shadow-offset:[object Object];left:0;background-color:rgb(242, 242, 242)"
                                />
                                <view
                                  style="0:[object Object];1:[object Object],"
                                >
                                  <view
                                    style="flex:1;flex-direction:column-reverse"
                                  >
                                    <view
                                      style="flex:1"
                                    >
                                      <rctsafeareaview>
                                        <rctscrollview
                                          contentcontainerstyle="[object Object]"
                                        >
                                          <view>
                                            <view>
                                              <view
                                                focusable="true"
                                                style="opacity:1"
                                              >
                                                <img
                                                  id="Forrest"
                                                  source="test-file-stub"
                                                  style="resize-mode:cover;height:150px;width:200px"
                                                />
                                                <text
                                                  style="text-align:center"
                                                >
                                                  Forrest
                                                </text>
                                              </view>
                                            </view>
                                            <view>
                                              <view
                                                focusable="true"
                                                style="opacity:1"
                                              >
                                                <img
                                                  id="Beach"
                                                  source="test-file-stub"
                                                  style="resize-mode:cover;height:150px;width:200px"
                                                />
                                                <text
                                                  style="text-align:center"
                                                >
                                                  Beach
                                                </text>
                                              </view>
                                            </view>
                                            <view>
                                              <view
                                                focusable="true"
                                                style="opacity:1"
                                              >
                                                <img
                                                  id="City"
                                                  source="test-file-stub"
                                                  style="resize-mode:cover;height:150px;width:200px"
                                                />
                                                <text
                                                  style="text-align:center"
                                                >
                                                  City
                                                </text>
                                              </view>
                                            </view>
                                            <view>
                                              <view
                                                focusable="true"
                                                style="opacity:1"
                                              >
                                                <img
                                                  id="Mountains"
                                                  source="test-file-stub"
                                                  style="resize-mode:cover;height:150px;width:200px"
                                                />
                                                <text
                                                  style="text-align:center"
                                                >
                                                  Mountains
                                                </text>
                                              </view>
                                            </view>
                                          </view>
                                        </rctscrollview>
                                      </rctsafeareaview>
                                    </view>
                                  </view>
                                </view>
                              </view>
                            </view>
                          </view>
                        </view>
                      </view>
                      <view
                        pointer-events="box-none"
                        style="0:[object Object];1:[object Object]"
                      />
                    </view>
                  </view>
                </view>
              </view>
            </view>
          </view>
        </view>
      </view>
    </view>
  </rncsafeareaprovider>
</view>
`;

exports[`<GalleryNav > Tab Nav Mock Tests MockTabNavigator <Hello > should render correctly 1`] = `
<view
  style="flex:1"
>
  <rncsafeareaprovider
    style="0:[object Object]"
  >
    <view
      pointer-events="box-none"
      style="0:[object Object];1:"
    >
      <view
        importantforaccessibility="auto"
        pointer-events="box-none"
      >
        <view
          pointer-events="box-none"
          style="position:absolute;left:0;right:0;top:0;bottom:0;z-index:0;opacity:1"
        >
          <view
            style="flex:1;shadow-opacity:0.85px;shadow-radius:0;shadow-offset:[object Object];background-color:rgb(255, 255, 255);border-bottom-color:rgb(216, 216, 216);shadow-color:rgb(216, 216, 216)"
          />
        </view>
        <view
          pointer-events="box-none"
          style="height:64px"
        >
          <view
            pointer-events="none"
            style="height:20px"
          />
          <view
            pointer-events="box-none"
            style="flex:1;flex-direction:row;align-items:center;justify-content:center"
          >
            <view
              pointer-events="box-none"
              style="margin-horizontal:16px;opacity:1"
            >
              <text
                accessibilityrole="header"
                numberoflines="1"
                style="font-size:17px;font-weight:600;color:rgb(28, 28, 30)"
              >
                MockedScreen
              </text>
            </view>
          </view>
        </view>
      </view>
    </view>
    <view
      style="flex:1"
    >
      <view
        pointer-events="box-none"
        style="position:absolute;left:0;right:0;top:0;bottom:0"
      >
        <view
          style="opacity:1"
        />
        <view
          gesturevelocityimpact="0.3"
          importantforaccessibility="auto"
          pointer-events="box-none"
          style="position:absolute;left:0;right:0;top:0;bottom:0"
          transitionspec="[object Object]"
        >
          <view
            pointer-events="box-none"
            style="flex:1"
          >
            <view
              style="flex:1;transform:[object Object],[object Object]"
            >
              <view
                pointer-events="none"
                style="position:absolute;shadow-radius:5px;shadow-color:#000;shadow-opacity:0.3px;top:0;bottom:0;width:3px;shadow-offset:[object Object];left:0;background-color:rgb(242, 242, 242)"
              />
              <view
                style="0:[object Object];1:[object Object],"
              >
                <view
                  style="flex:1;flex-direction:column-reverse"
                >
                  <view
                    style="flex:1"
                  >
                    <view
                      style="flex:1;align-items:center;justify-content:center"
                    >
                      <text
                        id="welcome-message"
                      >
                        Hello World
                      </text>
                    </view>
                  </view>
                </view>
              </view>
            </view>
          </view>
        </view>
      </view>
    </view>
  </rncsafeareaprovider>
</view>
`;
